services/worker-manager/src/providers/azure.js

```diff
@@ -387,19 +383,229 @@ class AzureProvider extends Provider {
 
   async scanPrepare() {
     this.seen = {};
+    this.errors = {};
+  }
+
+  /**
+   * Checks the status of ongoing Azure operations
+   * Returns true if the operation is in progress, false otherwise
+   *
+   * op: a URL for tracking the ongoing status of an Azure operation
+   * errors: a list that will have any errors found for that operation appended to it
+   */
+  async handleOperation({op, errors}) {
+    let req, resp;
+    try {
+      // TODO: we could be smarter about getting operation status
+      // see here: https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/async-operations
+      req = new msRestJS.WebResource(op, 'GET');
+      // sendLongRunningRequest polls until finished but this is just reading
+      // the status of an operation so it shouldn't block long
+      // it's ok if we hit an error here, that will trigger resource teardown
+      resp = await this._enqueue('opRead', () => this.restClient.sendLongRunningRequest(req));
+    } catch (err) {
+      // Rest API has different error semantics than the SDK
+      if (_.has(resp, 'status') && resp.status === 404) {
+        // operation not found because it has either expired or does not exist
+        // nothing more to do
+        return false;
+      }
+      throw err;
+    }
+
+    let body = resp.parsedBody;
+    if (body) {
+      // status is guaranteed to exist if the operation was found
+      if (body.status === 'InProgress') {
+        return true;
+      }
+      if (body.error) {
+        errors.push({
+          kind: 'operation-error',
+          title: 'Operation Error',
+          description: body.error.message,
+          extra: {
+            code: body.error.code,
+          },
+          notify: this.notify,
+          WorkerPoolError: this.WorkerPoolError,
+        });
+      }
+    }
+
+    return false;
+  }
+
+  /**
+   * provisionResource generically provisions individual resources
+   * Handles cases where:
+   *  we have not yet created a resource and need to create one,
+   *  we have requested a resource but it is not ready,
+   *  we have a resource ready to go
```

Comment:
I think these cases are identified by different values of `typeData.id`, `.name`, and `.operation`, but I'm not quite clear which is which -- could that be described here?

Reply:
Added a bit more detail.